source-azure-blob-storage/main.go

```diff
@@ -0,0 +1,355 @@
+package main
+
+import (
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/bloberror"
+	"github.com/estuary/connectors/filesource"
+	"github.com/estuary/flow/go/parser"
+	pf "github.com/estuary/flow/go/protocols/flow"
+)
+
+type config struct {
+	Credentials        credentials    `json:"credentials"`
+	StorageAccountName string         `json:"storageAccountName"`
+	Parser             *parser.Config `json:"parser"`
+	MatchKeys          string         `json:"matchKeys,omitempty"`
+	Advanced           advancedConfig `json:"advanced"`
+}
+type advancedConfig struct {
+	AscendingKeys bool `json:"ascendingKeys,omitempty"`
+}
+
+type credentials struct {
+	ContainerName       string `json:"containerName"`
+	AzureClientID       string `json:"azureClientID"`
+	AzureClientSecret   string `json:"azureClientSecret"`
+	AzureTenantID       string `json:"azureTenantID"`
+	AzureSubscriptionID string `json:"azureSubscriptionID"`
+	ConnectionString    string `json:"connectionString"`
+}
+
+func (c config) Validate() error {
+	if c.Credentials.ConnectionString != "" {
+		return nil
+	}
+	var requiredProperties [][]string
+	if c.Credentials.ConnectionString == "" {
+		requiredProperties = [][]string{
+			{"AzureTenantID", c.Credentials.AzureTenantID},
+			{"AzureClientID", c.Credentials.AzureClientID},
+			{"AzureClientSecret", c.Credentials.AzureClientSecret},
+			{"AzureSubscriptionID", c.Credentials.AzureSubscriptionID},
+			{"StorageAccountName", c.StorageAccountName},
+			{"ContainerName", c.Credentials.ContainerName},
+		}
+	} else {
```

Comment:
We never will reach this `else` because of the check on the line 41

Reply:
True, just fixed.